### DISCUSSION TITLE
Reading configuration from YAML

### DISCUSSION THREAD

**Discussion by marbx** [wixtoolset/issues#6653] [Questions] [answered]
An application uses a YAML file (from a well-known location) for configuration.
We wrote a C# CustomAction to read and write YAML, the WiX GUI presents parts of the configuration and the user can modify them.

Now need arose to read a YAML file from a user chosen location. 

I believe we must call the C# CustomAction before the `InstallUISequence`, and therefore cannot react to a user file selected within the GUI.

The [XMFile](https://wixtoolset.org/documentation/manual/v3/xsd/util/xmlfile.html) element reads a file which may be selected just a window before.

Is it of common interest to create a similar `YAMLFile` element?
Do we have other options?

**Comment by robmen** [comment, marked as answer]
It'd be cool if you created a WixExtension to edit YAML files for WiX v4. WiX v4 will make it much easier to reference other extensions so it wouldn't need to be integrated in the WiX Toolset itself.

**Reply by marbx** [reply]
Thank you for your affirmation, I will come back with questions.
